### PR TITLE
Add optional List<InlineSpan> parameter to BarTooltipItem, LineTooltipItem, ScatterTooltipItems for enhanced styling.

### PR DESCRIPTION
**Pull request by jrc2139**
redo of #554 

I expanded this to include the LineTooltipItem and ScatterTooltipItem because it was the same process as the BarTooltipItem.

### CODE REVIEW

**Comment by jrc2139** [comment]
not sure which branch I should merge into, so I set it to `dev-nullsafety`

**Comment by imaNNeoFighT** [comment]
Please create your branch from `dev`.
`dev-nulllsafety` has been merged before.

**Comment by imaNNeoFighT** [comment]
Does it fix #294 ?

**Comment by jrc2139** [comment]
> Does it fix #294 ?

yeah i think it does because i was wanted that solution as well.  i changed the parameter from `List<TextSpan>` to `List<InlineSpan>` because now we could pass `WidgetSpan`s also.  However, the painter freaks out null dimensions when i tried to pass a simple `Icon` to it.  I dont have a lot of experience with painting so maybe thats an easy fix.

**Comment by imaNNeoFighT** [comment]
As we need more investigation for InlineSpan. I suggest reverting it to TextSpan.
later we can check it out again.

File: lib/src/chart/radar_chart/radar_chart.dart [hunk 1]
```diff
@@ -18,10 +18,10 @@ class RadarChart extends ImplicitlyAnimatedWidget {
     Duration swapAnimationDuration = const Duration(milliseconds: 150),
     Curve swapAnimationCurve = Curves.linear,
   }) : super(
-      key: key,
-      duration: swapAnimationDuration,
-      curve: swapAnimationCurve,
-  );
+          key: key,
```
Comment (imaNNeoFighT) [inline lib/src/chart/radar_chart/radar_chart.dart:21]: Please revert this change.

Comment (jrc2139) [reply lib/src/chart/radar_chart/radar_chart.dart:21]: it was causing a failure on the pipeline

Comment (imaNNeoFighT) [reply lib/src/chart/radar_chart/radar_chart.dart:21]: I'm pretty sure that now dev is succeeding in the pipeline.
Please rebase your branch with dev again.
And also please revert InlineSpan to TextSpan

Comment (jrc2139) [reply lib/src/chart/radar_chart/radar_chart.dart]: ok all set!

**Comment by imaNNeoFighT** [comment]
Thanks for your contribution!
Hope to see more from you.

**Comment by flipperji** [comment]
> not sure which branch I should merge into, so I set it to `dev-nullsafety`

Hi ,bro.  I tried branch improvement/tooltipitem-inlinespan-null-safety , it's not work ,nothing shows on tips
5B3595FA-F865-493D-9BD7-795FF49CA6E1
Simple code like the bottom , Is there anything wrong?
WX20210706-165405@2x

**Comment by jrc2139** [comment]
> > not sure which branch I should merge into, so I set it to `dev-nullsafety`
> 
> Hi ,bro. I tried branch improvement/tooltipitem-inlinespan-null-safety , it's not work ,nothing shows on tips
> 5B3595FA-F865-493D-9BD7-795FF49CA6E1
> Simple code like the bottom , Is there anything wrong?
> WX20210706-165405@2x

Broo.. `WidgetSpan` is not supported and requires a lot of modifications to the Paint process.  Feel free to give it a shot though!

**Comment by flipperji** [comment]
> > > not sure which branch I should merge into, so I set it to `dev-nullsafety`
> > 
> > 
> > Hi ,bro. I tried branch improvement/tooltipitem-inlinespan-null-safety , it's not work ,nothing shows on tips
> > 5B3595FA-F865-493D-9BD7-795FF49CA6E1
> > Simple code like the bottom , Is there anything wrong?
> > WX20210706-165405@2x
> 
> Broo.. `WidgetSpan` is not supported and requires a lot of modifications to the Paint process. Feel free to give it a shot though!
Okay , Thank you
